### PR TITLE
More toStringDeep and toString tests.

### PR DESCRIPTION
**Pull request by jacob314**
These tests are a bit ugly but they've helped me catch a lot of edge cases in the pending
toStringDeep refactor I'm working on.

### CODE REVIEW

File: packages/flutter/test/painting/box_painter_test.dart [hunk 1]
```diff
@@ -54,6 +66,11 @@ void main() {
     expect(border4.left.width, equals(2.0));
   });
 
+  test('Border toString test', () {
+    expect(new Border.all(width: 4.0).toString(), equals(
+        'Border(BorderSide(Color(0xff000000), 4.0, BorderStyle.solid), BorderSide(Color(0xff000000), 4.0, BorderStyle.solid), BorderSide(Color(0xff000000), 4.0, BorderStyle.solid), BorderSide(Color(0xff000000), 4.0, BorderStyle.solid))'));
```
Comment (Hixie) [inline packages/flutter/test/painting/box_painter_test.dart]: nit: apply flutter style to the parens

File: packages/flutter/test/painting/box_painter_test.dart [hunk 2]
```diff
@@ -130,4 +151,16 @@ void main() {
...
+  test('LinearGradient toString', () {
+    expect(const LinearGradient(
+      begin: FractionalOffset.topLeft,
+      end: FractionalOffset.bottomLeft,
+      colors: const <Color>[
+        const Color(0x33333333),
+        const Color(0x66666666),
+      ],
+    ).toString(),
+        equals('LinearGradient(FractionalOffset(0.0, 0.0), FractionalOffset(0.0, 1.0), [Color(0x33333333), Color(0x66666666)], null, TileMode.clamp)'));
```
Comment (Hixie) [inline packages/flutter/test/painting/box_painter_test.dart]: ditto

File: packages/flutter/test/painting/flutter_logo_test.dart [hunk 3]
```diff
@@ -63,4 +63,11 @@ void main() {
     logo = FlutterLogoDecoration.lerp(start, end, 0.5);
     expect(logo.style, end.style);
   });
+
+  test('FlutterLogoDecoration toString', () {
+    expect(start.toString(), equals(
+        'FlutterLogoDecoration(Color(0xff000000)/Color(0xffffffff) on Color(0xffd4f144), FlutterLogoStyle.stacked)'));
+    expect(FlutterLogoDecoration.lerp(null, end, 0.5).toString(), equals(
+        'FlutterLogoDecoration(Color(0xffffffff)/Color(0xff000000) on Color(0xff81d4fa), FlutterLogoStyle.stacked, transition -1.0:0.5)'));
```
Comment (Hixie) [inline packages/flutter/test/painting/flutter_logo_test.dart]: ditto

File: packages/flutter/test/painting/text_style_test.dart [hunk 4]
```diff
@@ -9,6 +9,11 @@ import 'package:test/test.dart';
 
 void main() {
   test("TextStyle control test", () {
+    expect(const TextStyle(inherit: false).toString(), equals(
+        'inherit: false'));
+    expect(const TextStyle(inherit: true).toString(), equals(
+        'inherit: true'));
```
Comment (Hixie) [inline packages/flutter/test/painting/text_style_test.dart]: ditto

File: packages/flutter/test/rendering/paragraph_test.dart [hunk 5]
```diff
@@ -198,7 +195,7 @@ void main() {
         '   ║   "I polished up that handle so carefullee\n'
         'That now I am the Ruler of the Queen\'s Navee!"\n'
         '   ╚═══════════\n'
-        '\n',
+        '\n'
```
Comment (Hixie) [inline packages/flutter/test/rendering/paragraph_test.dart]: ?

File: packages/flutter/test/rendering/slivers_test.dart [hunk 6]
```diff
@@ -13,8 +13,39 @@ void main() {
     final RenderViewport root = new RenderViewport(
       offset: new ViewportOffset.zero(),
     );
+    // TODO(jacobr): remove extra trailing line break.
+    expect(root, isNot(hasAGoodToStringDeep));
```
Comment (Hixie) [inline packages/flutter/test/rendering/slivers_test.dart]: wait, why don't we want it to be correct?

File: packages/flutter/test/rendering/slivers_test.dart [hunk 7]
```diff
@@ -538,4 +655,17 @@ void main() {
...
+    // TODO(jacobr): cleanup the spurious trailing commas from the toString for
+    // SliverGeometry.
+    expect(
+        const SliverGeometry().toString(),
+        equals('SliverGeometry(scrollExtent: 0.0, hidden, maxPaintExtent: 0.0, )'));
+    expect(
+        const SliverGeometry(scrollExtent: 100.0, paintExtent: 50.0, layoutExtent: 20.0, visible: true).toString(),
+        equals('SliverGeometry(scrollExtent: 100.0, paintExtent: 50.0, layoutExtent: 20.0, maxPaintExtent: 0.0, )'));
+    expect(const SliverGeometry(scrollExtent: 100.0, paintExtent: 0.0, layoutExtent: 20.0).toString(),
+        equals('SliverGeometry(scrollExtent: 100.0, hidden, layoutExtent: 20.0, maxPaintExtent: 0.0, )'));
```
Comment (Hixie) [inline packages/flutter/test/rendering/slivers_test.dart]: nit: flutter style

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/test/painting/box_painter_test.dart (hunk 1 above)
Comment (jacob314) [inline packages/flutter/test/painting/box_painter_test.dart]: done

File: packages/flutter/test/painting/box_painter_test.dart (hunk 2 above)
Comment (jacob314) [inline packages/flutter/test/painting/box_painter_test.dart]: done

File: packages/flutter/test/painting/flutter_logo_test.dart (hunk 3 above)
Comment (jacob314) [inline packages/flutter/test/painting/flutter_logo_test.dart]: done

File: packages/flutter/test/rendering/paragraph_test.dart (hunk 5 above)
Comment (jacob314) [inline packages/flutter/test/rendering/paragraph_test.dart]: Added back. Merge error going between my branches.

File: packages/flutter/test/rendering/slivers_test.dart (hunk 6 above)
Comment (jacob314) [inline packages/flutter/test/rendering/slivers_test.dart]: We want it to be correct. The TODO on the previous line and the test expectation
will be flipped when I next land a cl that removes the extra line.

File: packages/flutter/test/rendering/slivers_test.dart (hunk 7 above)
Comment (jacob314) [inline packages/flutter/test/rendering/slivers_test.dart]: done

**Pull request merged**